Disable 802.3az Energy Efficient mode

robi251 opened the pull request:
Energy efficient mode is the feature of the DM8806 described in the EEE 802.3az Energy Efficient for reducing power consumption. For unknown reason it seams to not working correctly with all endpoints and sometimes the network randomly fails when this mode is on. Thus it is now possible to turn off it in compile time by KConfig option:
PHY_DM8806_ENABLE_ENERGY_EFFICIENT_MODE if in case of network problems

ZiadElhanafy committed:
drivers: gic: Add multiple GIC redistributors regions support

For GIC multiple views feature support, all GIC Re-distributor's
GICR_TYPER.last will be set. Because configuration view-0 can
assign non-contiguous CPUs to views other than 0, in this case
the GIC Redistributors' registers won't seem contiguous.

So the GIC driver should cope with multiple sets of redistributors
like multi-chip scenarios. In this patch we add multiple GIC
redistributor regions support in GIC redistributor iteration.

For more information, refer to the Multi view subsection
in the GIC Technical Reference Manual.
For example:
https://developer.arm.com/documentation/101516/0400/Operation-of-GIC-700/Multi-view

Signed-off-by: Ziad Elhanafy <ziad.elhanafy@arm.com>
Signed-off-by: Robert Slawinski <robert.slawinski1@gmail.com>

ttmut committed:
drivers: serial: uart_max32: Add async mode support

This commit adds asynchronous mode support to MAX32 UART driver. Each
direction uses a single DMA channel that is assigned in devicetree
configuration.

Asynchronous mode also depends on interrupts to refresh receive
timeouts.

Signed-off-by: Tahsin Mutlugun <Tahsin.Mutlugun@analog.com>

ttmut committed:
tests: drivers: uart: uart_async_api: Enable MAX32 boards

Support following boards in asynchronous UART tests:

- MAX32655EVKIT
- MAX32655FTHR
- MAX32666EVKIT
- MAX32666FTHR
- MAX32670EVKIT
- MAX32690EVKIT

Signed-off-by: Tahsin Mutlugun <Tahsin.Mutlugun@analog.com>

ttmut committed:
drivers: serial: uart_max32: Use cache to workaround DMA limitation

When using asynchronous API, transfer will fail if the source buffer is
located in a region that cannot be accessed by DMA. This could happen
when a buffer is declared const, and placed in flash memory, for
example.

Workaround this problem by loading the data into a set of temporary
caches before passing them to DMA.

Signed-off-by: Tahsin Mutlugun <Tahsin.Mutlugun@analog.com>

congnguyenhuu committed:
drivers: flash: add NXP S32 QSPI HyperFlash driver

Add support HyperFlash memory devices on a NXP S32 QSPI bus.
This driver uses a fixed LUT configuration that defined in HAL RTD
HyperFlash driver.
Driver allows to read, write and erase HyperFlash devices.

Signed-off-by: Cong Nguyen Huu <cong.nguyenhuu@nxp.com>

Jaakko-Bit committed:
drivers: sensor: ti: tmp435: added driver for TMP435

The TMP435 is a remote temperature sensor monitor
with a built-in local temperature sensor.

Signed-off-by: Jaakko Rautiainen <jaakko.rautiainen@bittium.com>

congnguyenhuu committed:
drivers: memc_nxp_s32_qspi: add support for s32ze

Add support QSPI secure flash protection (SFP)

Signed-off-by: Cong Nguyen Huu <cong.nguyenhuu@nxp.com>

congnguyenhuu committed:
samples: enable flash samples for s32z board

Enable flash samples for s32z board

Signed-off-by: Cong Nguyen Huu <cong.nguyenhuu@nxp.com>

congnguyenhuu committed:
tests: enable flash tests for s32z board

Enable flash tests for s32z board

Signed-off-by: Cong Nguyen Huu <cong.nguyenhuu@nxp.com>

congnguyenhuu committed:
boards: s32z2xx: enable flash controller QSPI

The on-board S26HS512T 512M-bit HyperFlash memory is connected to
the QSPI controller port A1.
This board configuration selects it as the default flash controller.

Signed-off-by: Cong Nguyen Huu <cong.nguyenhuu@nxp.com>

congnguyenhuu committed:
drivers: flash_nxp_s32: create common source code

Create common source code to use for supporting HyperFlash.

Rename 'FLASH_NXP_S32_QSPI_NOR_SFDP_RUNTIME' to
'FLASH_NXP_S32_QSPI_SFDP_RUNTIME' as a common kconfig.

Add the 'max-program-buffer-size' property to use for
setting memory pageSize, instead of using
'CONFIG_FLASH_NXP_S32_QSPI_LAYOUT_PAGE_SIZE' for setting.

Add the 'write-block-size' propertyto use for setting
the number of bytes used in write operations, it also
uses to instead of the 'memory-alignment' property.

Signed-off-by: Cong Nguyen Huu <cong.nguyenhuu@nxp.com>

Commit:
boards: teensy4: Update board documentation

- insert various additional information content
- use tabs to distinguish board versions
- west flash is now usable for flashing
- show steps for using USB-console
- move images to img subfolder

Signed-off-by: Bernhard Krämer <bdkrae@gmail.com>

congnguyenhuu committed:
drivers: memc_nxp_s32_qspi: change DT_REG_ADDR to DT_REG_ADDR_RAW

Following the commit f98fde07b30, DT_REG_ADDR now expands with a 'U'
suffix as an unsigned value. However, for compatibility with IS_EQ,
a raw value without any suffix is required. Therefore, this update is
necessary.

Signed-off-by: Cong Nguyen Huu <cong.nguyenhuu@nxp.com>

anobli committed:
samples: net: openthread: Add a sample using OpenThread CoAP API

In order to show how to use OpenThread and CoAP, add an application.
This example could be build to run a client or server.
The server could expose LEDs and buttons and client could get their
state and set the LEDs state.

The network is created automatically using the network key predefined
in the config. The goal is to make the example simple by removing
the commisionning and joinning process.

If a client application has a button, it could use it to toggle the first
LED on any boards running the server application.

Signed-off-by: Alexandre Bailon <abailon@baylibre.com>

anobli committed:
samples: net: openthread: coap: Add shell commands

This adds shell commands to easily get or set LED.

Signed-off-by: Alexandre Bailon <abailon@baylibre.com>

tswaehn committed:
drivers: serial: sy1xx add support for pinctrl

Add pin ctrl to the serial driver.

Signed-off-by: Sven Ginka <s.ginka@sensry.de>

tswaehn committed:
board: ganymed_bob_sy120: add uart pinctrl

Add pin definitions for the available uarts
for the ganymed_bob_sy120 boards.

Signed-off-by: Sven Ginka <s.ginka@sensry.de>

nhutnguyenkc committed:
drivers: serial: Add interrupt-driven API support for RZ/G3S

This commit is to support UART interrupt-driven API for Renesas RZ/G3S.

Signed-off-by: Nhut Nguyen <nhut.nguyen.kc@renesas.com>

nhutnguyenkc committed:
boards: renesas: rzg3s_smarc: Fix typo in RZ/G3S-SMARC Kconfig

Fix typo in RZ/G3S-SMARC Kconfig

Signed-off-by: Nhut Nguyen <nhut.nguyen.kc@renesas.com>

tswaehn committed:
drivers: gpio: add gpio support for sy1xx

Add gpio support for sensry soc sy1xx.

Signed-off-by: Sven Ginka <s.ginka@sensry.de>

mathieuchopstm committed:
dts: arm: st: wb0: add TRNG node

Add Device Tree nodes corresponding to TRNG of STM32WB0 series SoCs.

Signed-off-by: Mathieu Choplain <mathieu.choplain@st.com>

nhutnguyenkc committed:
doc: Add troubleshooting for RZ/G3S-SMARC

Add troubleshooting for HW resource conflicts between Linux and Zephyr

Signed-off-by: Nhut Nguyen <nhut.nguyen.kc@renesas.com>

mathieuchopstm committed:
tests: entropy: api: add overlay for Nucleo-WB05KZ

Add overlay to allow testing of RNG driver on STM32WB0 hardware.

Signed-off-by: Mathieu Choplain <mathieu.choplain@st.com>

mathieuchopstm committed:
drivers: entropy: stm32: add support for TRNG without interrupt line

Add support for TRNG peripherals that lack interrupt lines in the STM32
entropy driver.

This enables usage of the TRNG of STM32WB05/06/07 SoCs with the driver.

Signed-off-by: Mathieu Choplain <mathieu.choplain@st.com>

mathieuchopstm committed:
dts: bindings: rng: add STM32 binding for IRQ-less RNG

Add a new binding for STM32 RNG instances without interrupt lines,
such as the one present in the STM32WB05/06/07 SoCs.

Signed-off-by: Mathieu Choplain <mathieu.choplain@st.com>

mathieuchopstm committed:
drivers: entropy: stm32: add support for STM32WB09 TRNG

Add support for the STM32WB09-specific TRNG IP in STM32 entropy driver.

Signed-off-by: Mathieu Choplain <mathieu.choplain@st.com>

mathieuchopstm committed:
tests: entropy: api: add overlay for Nucleo-WB09KE

Add overlay to allow testing of RNG driver on STM32WB0 hardware.

Signed-off-by: Mathieu Choplain <mathieu.choplain@st.com>

mathieuchopstm committed:
tests: entropy: api: add overlay for Nucleo-WB07CC

Add overlay to allow testing of TRNG on STM32WB07.

Signed-off-by: Mathieu Choplain <mathieu.choplain@st.com>

MarkWangChinese committed:
usb: udc: fix noncache usb transfer buffer

CONFIG_UDC_BUF_FORCE_NOCACHE depend on NOCACHE_MEMORY && DCACHE in Kconfig,
so it can only be true when the platform supports cache.
The controller driver can imply the value of CONFIG_UDC_BUF_FORCE_NOCACHE,
like UDC_NXP_EHCI. If CONFIG_UDC_BUF_FORCE_NOCACHE is true, then all the
usb transfer buffer should be in noncache section.

Signed-off-by: Mark Wang <yichang.wang@nxp.com>

MaochenWang1 committed:
net: ip: net_pkt: only reserve L2 header for TX case

Only reserve L2 header for TX case when allocating net buffer,
as for RX case, all the received headers are already in linear
buffer when the driver receives the data, and reserve extra L2
header for RX case may exceed the default buffer size.

Signed-off-by: Maochen Wang <maochen.wang@nxp.com>

wmrsouza committed:
drivers: display: ssd1306: add support to ssd1309

SSD1309 support added to SSD1306 display driver

Signed-off-by: Marcio Ribeiro <wmrsouza@hotmail.com>

JordanYates committed:
runners: openocd: support `--erase`

Add support for the `--erase` option on `west flash`, for boards that
specify at least one `--cmd-erase` option.

Signed-off-by: Jordan Yates <jordan@embeint.com>

JordanYates committed:
boards: openocd: `--cmd-erase` for STM32 series

Add the appropriate mass erase command for STM32 SoC families.
Family to command mapping taken from:
  https://openocd.org/doc/html/Flash-Commands.html

Signed-off-by: Jordan Yates <jordan@embeint.com>

Commit:
boards: nxp: add RT1060 EVKC support

MIMXRT1060 EVK rev C initial support files.

Signed-off-by: David Leach <david.leach@nxp.com>

Commit:
samples: settings: Enable NVS backend on mimxrt1020_evk board

The tests for settings will not work properly with just provided config
file for mimxrt1020:
west build -p always -b mimxrt1020_evk samples/subsys/settings

as there the settings_save_one() would return -2 (-ENOENT) as by default
the CONFIG_SETTINGS_NONE is defined for mimxrt1020 configuration.

For proper operation (i.e. to have a memory medium to store settings
values) the NVS backend needs to be enabled.

Signed-off-by: Lukasz Majewski <lukma@denx.de>

Vge0rge committed:
tests: arm_irq_vector_table: Disable bellboard for nRF54/nRF92

This test uses the bellboard interrupts for the application
and the radio core builds on nRF54H20 and nRF9280.
Since it uses the bellboard interrupts it makes sense to ensure
that bellboard is disabled in device tree to avoid runtime issues.

This is preparation work, bellboard is planned to be enabled
by default later and this makes sure that the test will
continue to work.

Signed-off-by: Georgios Vasilakis <georgios.vasilakis@nordicsemi.no>

Commit:
spi_mcux_lpspi: Determine lpflexcomm by inst

Determine proper irq setup depending whether the spi is wrapped in an
lpflexcomm or not on an instance basis since some platforms have some
wrapped and some not.

Signed-off-by: Declan Snyder <declan.snyder@nxp.com>

rysiof committed:
arch: Changes MPU alignment to 128 when FPU_SHARING and MPU_STACK_GUARD

With our builds for Arm M7 we are selecting FPU_SHARING and
MPU_STACK_GUARD. For that setup, we have
ARM_MPU_REGION_MIN_ALIGN_AND_SIZE=64 and
MPU_STACK_GUARD_MIN_SIZE_FLOAT=128. So worst case scenario,
we are going to have stack that is 64-byte aligned and
128-byte guard at the bottom of the buffer.

Fixes: #83714

Signed-off-by: Maciej Kusio <rysiof@gmail.com>

nordicjm committed:
cmake: modules: Prefer imgtool being in MCUboot directory

Changes the priority of imgtool so that the preferred version is
the one inside of the MCUboot directory

Signed-off-by: Jamie McCrae <jamie.mccrae@nordicsemi.no>

EmilioCBen committed:
boards: nxp: frdm_mcxw71: Enable MCXW71 I2C Loopback

Enable and test I2C loopback with i2c_target_api

Signed-off-by: Emilio Benavente <emilio.benavente@nxp.com>

d3zd3z committed:
kernel: poll: Allow 0 events for submitted work

Change:
    commit cc6317d7ac30479db060f60c66bb48a12b2d0be6
    Author: Jukka Rissanen <jukka.rissanen@linux.intel.com>
    Date:   Fri Nov 1 14:03:32 2019 +0200

        kernel: poll: Allow 0 event input

Allows `k_poll` to be user with 0 events, which is useful for allowing just
a sleep without having to create artificial events.

Allow the same for `k_work_submit_to_queue()` and `k_work_submit()`.

Signed-off-by: David Brown <david.brown@linaro.org>

npitre committed:
kernel/pipe: code flow cleanup

Simplify the logic, avoid repeated conditionals, avoid superfluous
scheduler calls, make the code more efficient and easier to read.

Signed-off-by: Nicolas Pitre <npitre@baylibre.com>

jilaypandya committed:
dts: bindings: stepper: use hyphens instead of underscore

This commit replaces hyphens with underscore in order to comply with
devicetree standards

Signed-off-by: Jilay Pandya <jilay.pandya@outlook.com>

JordanYates committed:
fs: allow enabling filesystems with `FILE_SYSTEM_LIB_LINK`

Update the Kconfig configuration to allow enabling the underlying
filesystems (e.g. exFAT), with only `FILE_SYSTEM_LIB_LINK` enabled.

Signed-off-by: Jordan Yates <jordan@embeint.com>

JordanYates committed:
fs: ext2: update `CONFIG_MAX_FILES` name

Add the `EXT2` prefix to `config MAX_FILES` to more appropriately
namespace the symbol.

Signed-off-by: Jordan Yates <jordan@embeint.com>

JordanYates committed:
tests: fs: lib_link: added

Add a test validating that Zephyr can compile with the 3 underlying
filesystems enabled with only `CONFIG_FILE_SYSTEM_LIB_LINK` enabled.

Signed-off-by: Jordan Yates <jordan@embeint.com>

kietavainen committed:
drivers: bluetooth: silabs: Add separate thread for BT Link Layer

The BT Link Layer needs to get runtime in a timely manner to keep
connections alive and handle other time-critical tasks. This is achieved by
adding a separate thread for it with a meta-IRQ priority, which means it
can preempt other threads. The driver also has an RX thread that passes HCI
messages from the controller to the host stack. This can be a lower
priority cooperative thread, as it doesn't have strict timing requirements.

Signed-off-by: Kalle Kietäväinen <kalle.kietavainen@silabs.com>

kietavainen committed:
drivers: bluetooth: silabs: Potentially discard unimportant HCI events

Certain HCI events, like advertising reports, are of less importance than
others. This takes the discardable buffer pool into use for such events.
When the system is flooded with advertising reports, discarding some of
them ensures that the system can still handle other events.

Signed-off-by: Kalle Kietäväinen <kalle.kietavainen@silabs.com>

tmon-nordic committed:
drivers: udc_dwc2: Fix large control write transfers

The DOEPTSIZ0 XferSize field is 7 bits long, meaning that maximum single
transfer can be 127 bytes long. Configure the control write (OUT)
transfers considering the XferSize field size to support transfers with
data stage larger than 127 bytes.

Signed-off-by: Tomasz Moń <tomasz.mon@nordicsemi.no>

npitre committed:
kernel/pipe: remove extra hardware timer query on thread wakeup

Dispense with the call to sys_timepoint_expired() by leveraging
swap_retval to distinguish between notifications and timeouts when
z_pend_curr() returns.

Signed-off-by: Nicolas Pitre <npitre@baylibre.com>

butok committed:
boards: mimxrt1180_evk: set LinkServer as the default runner

Sets LinkServer as the default runner for the mimxrt1180-evk board,
as the board is configured for CMSIS-DAP by default.

Signed-off-by: Andrej Butok <andrey.butok@nxp.com>

tmon-nordic committed:
drivers: udc_dwc2: Recover after STALLed OUT Data Stage

Prepare buffer to receive SETUP data on OUT endpoint 0 after endpoint
halt. This solves the issue where the device would no longer process any
control transfers after the first failed transfer with too large OUT
Data Stage (when processing failed due to data stage buffer allocation
failure).

Signed-off-by: Tomasz Moń <tomasz.mon@nordicsemi.no>

gatzka committed:
modbus: serial: check return value of uart_irq_callback_user_data_set

uart_irq_callback_user_data_set return non-zero if the callback function
can't be set. This might happen if you forgot to enable
CONFIG_UART_INTERRUPT_DRIVEN.

Signed-off-by: Stephan Gatzka <stephan.gatzka@gmail.com>

Thalley committed:
Bluetooth: GATT: Change get_handle function of find_by_uuid

bt_gatt_find_by_uuid used bt_gatt_attr_value_handle but that
function only works to get the value handle of a characteristic
declaration, i.e. if the UUID is not BT_UUID_GATT_CHRC then it
would always return handle = 0. This meant that
bt_gatt_find_by_uuid would always use handle = 0 as the starting
handle for non-BT_UUID_GATT_CHRC attributes, instead of the handle
of the provided attr.

This was not an issue for any UUIDs that may only exist once on a
GATT server, which is most UUIDs, but for UUIDs like the
BT_UUID_TBS_* UUIDs that may be multiple instances of, it would
always return the first attribute rather than the one starting
from the provided start attr.

This commit also ensures that we do not overflow the `end_handle`
when adding 2 uint16_t values.

Signed-off-by: Emil Gydesen <emil.gydesen@nordicsemi.no>

gatzka committed:
modbus: initialize return variable.

Currently, when disabling CONFIG_MODBUS_FP_EXTENSIONS,return variable
'err'  is not initialized (at least not for debug builds) but later on
checked.

Signed-off-by: Stephan Gatzka <stephan.gatzka@gmail.com>

Commit:
i2c_mcux_lpspi: LPflexcomm determine by inst

Determine if lpflexcomm wrapped lpi2c by instance and connect
irq differently dependending on that to support platforms with
both flexcomm wrapped and unwrapped lpi2c's.

Signed-off-by: Declan Snyder <declan.snyder@nxp.com>

Commit:
uart_mcux_lpuart: LPFlexcomm inst based determine

Since some platforms may have some lpuart that are wrapped in lpflexcomm
and some that are not, then change the init code to determine how to
connect the interrupt on an instance basis.

Signed-off-by: Declan Snyder <declan.snyder@nxp.com>

npitre committed:
kernel/pipe: implement direct-to-pending-readers data copy

If there are pending readers, it is best to perform a single data copy
directly into their final destination buffer rather than doing one copy
into the ring buffer just to immediately copy the same data out of it.

Incidentally, this allows for supporting pipes with no ring buffer at all.

The pipe implementation being deprecated has a similar capability so better
have it here too.

Signed-off-by: Nicolas Pitre <npitre@baylibre.com>

tswaehn committed:
dts: sy1xx: add gpio support

With this commit we add the gpio functionality to
the sensry soc sy1xx.

Signed-off-by: Sven Ginka <s.ginka@sensry.de>

djiatsaf-st committed:
drivers: dma: cyclic dma handling

keeping the stream busy when cyclic DMA is used.

Signed-off-by: Fabrice DJIATSA <fabrice.djiatsa-ext@st.com>

djiatsaf-st committed:
drivers: dma: add linked list in dmau5 driver

- Init linked list node and structure
- configure source and destination address
- Setting gpdma request mode
- Configure linked list node and structure for
continous transfer, enable half transfert irq.

Signed-off-by: Fabrice DJIATSA <fabrice.djiatsa-ext@st.com>
co-authored-by: Lubos Koudelka <lubos.koudelka@st.com>

djiatsaf-st committed:
samples: boards: st: uart: circular_dma: update circular dma sample

Add the nucleo_wba55cg board, which implements continuous transfer
with GPDMA and a linked list.

Signed-off-by: Fabrice DJIATSA <fabrice.djiatsa-ext@st.com>

FlorianWeber1018 committed:
tests: drivers: build_all: sensors: added mlx90394

added mlx90394 to the testcase so that it is in build.

Signed-off-by: Florian Weber <Florian.Weber@live.de>

tmon-nordic committed:
drivers: udc_dwc2: Fix multipart DMA OUT transfers

DMA transfers are supposed to write to buffer tail. Use the proper
pointer to make multipart DMA transfers actually write the data to the
intended location.

The issue was observed on control write transfers where the OUT Data
Stage was at least 128 bytes (because endpoint 0 transfer width is
limited to 7 bits).

The issue is unlikely to happen on non-control transfers because the
transfer size width is at least 11 bits (at most 19 bits) and packet
size counter is at least 4 bits (at most: 10 bits) which means that
at least 2048 byte transfer spanning at least 15 packets (or at least
524288 byte spanning 1023 packets for 19 bits transfer size counter
and 10 bits packet counter) is required to necessitate multipart DMA.

Signed-off-by: Tomasz Moń <tomasz.mon@nordicsemi.no>

npitre committed:
tests: app_kernel: restore the PIPE_NOBUFF variant

... now that the new pipe implementation supports it.

Signed-off-by: Nicolas Pitre <npitre@baylibre.com>

JordanYates committed:
fs: ext2: fix uninitialized variable

The compiler is unaware of the `block_size > 0` precondition, so it
outputs warnings of uninitialized `reclen`. Explicitly initialize the
variable to suppress this.

Signed-off-by: Jordan Yates <jordan@embeint.com>

jilaypandya committed:
docs: migration_guide_4_1: migration guide entry for renamed properties

add migration guide for following renamed variables
 - ``en_spreadcycle`` to ``en-spreadcycle``
- ``i_scale_analog`` to ``i-scale-analog``
- ``index_optw``to ``index-otpw```
- ``ìndex_step`` to ``index-step``
- ``internal_rsense`` to ``internal-rsense``
- ``lock_gconf`` to ``lock-gconf``
- ``mstep_reg_select`` to ``mstep-reg-select``
- ``pdn_disable`` to ``pdn-disable``
- ``poscmp_enable`` to ``poscmp-enable``
- ``test_mode`` to ``test-mode``

Signed-off-by: Jilay Pandya <jilay.pandya@outlook.com>

pdgendt committed:
linker: iterable_section: Allow numeric sorting up to 5 numeric characters

This allows using a uint16_t value to sort iterable sections.

Signed-off-by: Pieter De Gendt <pieter.degendt@basalte.be>

pdgendt committed:
tests: linker: iterable_sections: Test sorting with 5 numeric characters

Update the iterable numeric RAM/ROM tests to include a numeric value with
5 numeric characters.

Signed-off-by: Pieter De Gendt <pieter.degendt@basalte.be>

felf-zhaw committed:
docs: samples: subsys: lorawan: class_a: update README

Updated the README for the LoRaWAN Class A example to explain limitations
when running multiple times. Provided guidance to resolve the issue.

Signed-off-by: Flavio Felder <felf@zhaw.ch>

nordic-babu committed:
samples: boards: nordic: system_off: fix sample regex.

Application won't wakeup without external trigger in
the analog comparator wakeup configuration.

Therefore removing post-wakeup lines from sample regex as they are not
expected to occur in default testing scenario (without external triggers).

Signed-off-by: Bartlomiej Buczek <bartlomiej.buczek@nordicsemi.no>

aescolar committed:
Bluetooth: Controller: Correct nrf54l15bsim timings

Correct the timings in the radio HAL for the simulated nRF54L15

Signed-off-by: Alberto Escolar Piedras <alberto.escolar.piedras@nordicsemi.no>

nordicjm committed:
mgmt: mcumgr: grp: img_mgmt: Fix unused label warning

Fixes an issue with a warning when compiling for direct xip mode
without revert

Signed-off-by: Jamie McCrae <jamie.mccrae@nordicsemi.no>

kica-z committed:
drivers: disk: sdmmc_subsys: stm32_sdmmc driver custom disk name support

In order to allow a custom disk name same as with the standard
sdmmc driver an additional device-tree property was introduced.

Signed-off-by: Carlo Kirchmeier <carlo.kirchmeier@zuehlke.com>

Martinhoff-maker committed:
driver: serial: split silabs series 2 and series 0/1 boards usart driver

Split the USART driver into separate implementations for Silabs Series 2
and Series 0/1 boards. This change improves maintainability (especially
with the support of pin-ctrl and clock-ctrl on series 2 boards).

Signed-off-by: Martin Hoff <martin.hoff@silabs.com>

avisconti committed:
drivers/sensor: ism330dhcx: toggle device_conf bit

Set ctrl9_xl.device_conf bit to 1 in ism330dhcx_init_chip()
prior to start device configuration, as stated in paragraph
9.20 of the datasheet.

Signed-off-by: Armando Visconti <armando.visconti@st.com>

rlubos committed:
tests: net: coap: Add test case for matching pending replies

Add test case for matching pending replies with received responses.
Cover corner cases that are failing with the current implementation.

Signed-off-by: Robert Lubos <robert.lubos@nordicsemi.no>

cvinayak committed:
Bluetooth: Controller: Fix regression in connection update

The connection event at the connection update instant was
skipped due to previous ticker node in use that is being
stopped was registering a relative occupied time that
overlapped with the new ticker node being started for
scheduling the connection with new interval.

Added mock interface for ticker_stop_abs() hence needed by
the Controller unit testing.

Signed-off-by: Vinayak Kariappa Chettimada <vich@nordicsemi.no>

butok committed:
boards: frdm_mcxn947: fix mcuboot partitions

Fixes frdm_mcxn947 MCUBoot partitions
that were broken by d8ab81329ef47b5d6b0723d24932148cfc3b380a

Signed-off-by: Andrej Butok <andrey.butok@nxp.com>

nrbrook committed:
cmake: Allow semver style EXTRAVERSION

cmake: Allow semver style EXTRAVERSION

Semver versions are commonly in the form X.X.X-a.1, X.X.X-beta.2, etc.
However, currently EXTRAVERSION cannot include ".". This change allows it.

Signed-off-by: Nick Brook <nrbrook@gmail.com>

cvinayak committed:
Bluetooth: Controller: Fix reset of drift_skip count

Fix reset of drift_skip count when an anchor point sync was
missed so that drift compensation is applied on the next
successful anchor point sync. This will improve connection
stability for short overlapping radio event intervals.

Signed-off-by: Vinayak Kariappa Chettimada <vich@nordicsemi.no>

Martinhoff-maker committed:
dts: arm: change usart binding of silabs series 2 boards

Make the Silabs series 2 boards use the new USART driver
"silabs,usart-uart".

Signed-off-by: Martin Hoff <martin.hoff@silabs.com>

jukkar committed:
net: ethernet: Only try ARP for IP packets

The work in Ethernet send in commit 2f10d7d81624
("net: ethernet: Set the ptype by the caller in send")
introduced a bug because we could try to do ARP resolving
for ARP packets too. This is clearly a wrong thing to do.
So before trying to do ARP resolving, make sure the the
packet type is IP packet.

Signed-off-by: Jukka Rissanen <jukka.rissanen@nordicsemi.no>

jukkar committed:
net: arp: Enhance debug messages

Print more data / debug information for ARP messages.
Also remove unnecessary "&" when printing IPv4 address.

Signed-off-by: Jukka Rissanen <jukka.rissanen@nordicsemi.no>

jukkar committed:
tests: net: arp: Remove the ARP cache clear from the tests

It was incorrectly introduced earlier and not really needed
because the test will pass without it.

Signed-off-by: Jukka Rissanen <jukka.rissanen@nordicsemi.no>

rlubos committed:
net: coap: Fix response matching algorithm

The algorithm for matching request with response was incorrect, which
could lead to false matches (for example if request had a token, and
piggybacked reply had no token but matching message ID only, that would
still be counted as a match).

This commit fixes it. The request/reply matching is implemented based on
RFC now, with separate conditions for piggybacked/separate responses.

Signed-off-by: Robert Lubos <robert.lubos@nordicsemi.no>

snematbakhsh committed:
soc: lowrisc: opentitan: Fix manifest sizing

The actual manifest is 1024 bytes, but we previously had padding bytes
due to the implementation of SECTION_FUNC(). The manifest is not
executable code so SECTION_VAR() is appropriate and produces a section of
the appropriate size.

Fixes: #82822

Signed-off-by: Shawn Nematbakhsh <shawn@rivosinc.com>

VynDragon committed:
drivers: timer: Harmonize mtime-based RISC-V timers

This commit replaces a bunch of ifdefs and bindings with a single
extensible binding, and makes all standard mtime system timers consistent.

Signed-off-by: Camille BAUD <mail@massdriver.space>

brandon-exact committed:
zbus: assert when inside an ISR and time out is not zero

Currently various zbus functions silently change the timeout
to zero  when inside an ISR. If a developer is not aware
of this it could lead to unexpected behaviour or a
publish/read failing.

Also in the zbus docs it states to only use a timeout of
zero when inside a ISR multiple times.

Signed-off-by: Brandon Allen <brandon.allen@exacttechnology.com>

VynDragon committed:
drivers: timer: Remove unused divider setter in RISC-V machine timer

This removes a unused function for a unused binding,
and is prelude to a refactoring

Signed-off-by: Camille BAUD <mail@massdriver.space>

VynDragon committed:
drivers: timer: run clang-format on RISC-V machine timer file

Clang-Format changes whitespacing and longer lines are now allowed

Signed-off-by: Camille BAUD <mail@massdriver.space>

fabiobaltieri committed:
i2c_shell: add device filtering

Add shell device name filtering using DEVICE_API_IS.

Signed-off-by: Fabio Baltieri <fabiobaltieri@google.com>

fabiobaltieri committed:
regulator_shell: add device filtering

Add shell device name filtering using DEVICE_API_IS.

Signed-off-by: Fabio Baltieri <fabiobaltieri@google.com>

fabiobaltieri committed:
dac_shell: add device filtering

Add shell device name filtering using DEVICE_API_IS.

Signed-off-by: Fabio Baltieri <fabiobaltieri@google.com>

npitre committed:
kernel/pipe: add missing calls to z_reschedule()

We are waking up threads but failed to let them run if they are
higher priority. Add missing calls to z_reschedule().

Also wake up all pending writers as we don't know how many there might
be. It is more efficient to wake them all when the ring buffer is full
before reading from it rather than waking them one by one whenever there is
more room in it.

Thanks to Peter Mitsis for noticing those issues.

Signed-off-by: Nicolas Pitre <npitre@baylibre.com>

rahul-arasikere committed:
boards: esp32s3_touch_lcd_1_28: Fix default flash partition

The default partition file included in the board definition is for 4MB
flash. However, this board has 16MB flash.

Signed-off-by: Rahul Arasikere <arasikere.rahul@gmail.com>

cvinayak committed:
Bluetooth: Controller: Fix uninitialized is_aborted in conn done event

Fix uninitialized is_aborted in connection done event.

Relates to commit cadef5a64fa5 ("Bluetooth: Controller:
Introduce BT_CTLR_PERIPHERAL_RESERVE_MAX").

Signed-off-by: Vinayak Kariappa Chettimada <vich@nordicsemi.no>

FlorianWeber1018 committed:
drivers: sensor: mlx90394: added driver

Added driver for Melexis MLX90394 magnetometer.

Signed-off-by: Florian Weber <Florian.Weber@live.de>

tpennors committed:
Bluetooth: fix old-style warning gcc

fix GCC warning old-style-declaration on bluetooth host/att.c

Signed-off-by: PENNORS Thomas <t.pennors@lacroix.group>

ndrs-pst committed:
dts: arm: nxp: s32: add `#address-cells` to interrupt provider

Add `#address-cells = <0>;` to interrupt provider nodes in
the NXP S32 device tree to resolve warnings: e.g.
Warning (interrupt_provider): /soc/interrupt-controller@47800000: Missing
Warning (interrupt_provider): /soc/siul2@40520000/eirq0@40520010: Missing

This ensures compliance with device tree specifications and
eliminates build warnings.

Signed-off-by: Pisit Sawangvonganan <pisit@ndrsolution.com>

Commit:
drivers: ethernet: Fix DSA tag size for KSZ8863.

Allow DSA_TAG_SIZE to be set based on the size determined by Kconfig
and the enabled switch hardware.

This fixes support for the KSZ8863 which also has a tail tag of
one byte.

Signed-off-by: Bas van Loon <s.r.vanloon@ziggo.nl>

npitre committed:
kernel/pipe: disable direct-to-readers copy with CONFIG_KERNEL_COHERENCE

Systems that enabled this option don't have their stacks in coherent
memory. Given our pipe_buf_spec is stored on the stack, and readers may
also have their destination buffer on their stack too, it is not worth
going to the trouble of supporting direct-to-readers copy with them.

Signed-off-by: Nicolas Pitre <npitre@baylibre.com>

fabiobaltieri committed:
drivers: drop few redundant device_is_ready

Drop few redundant device_is_ready for functions that are only used as
argument to shell_device_filter, as shell_device_filter checks for that
alrady.

Suggested-by: Yishai Jaffe <yishai1999@gmail.com>
Signed-off-by: Fabio Baltieri <fabiobaltieri@google.com>

weeTike committed:
ipc: ipc_service: Null terminate RPMsg endpoint names

This fixes the compiler warning -Wstringop-truncation in
version 13 of gcc.

Signed-off-by: Timothy Keys <timothy.keys@nordicsemi.no>

JordanYates committed:
west.yml: update `nrf_wifi` to ignore binary blobs

Update the `nrf_wifi` repository so that git ignores binary blobs
fetched by `west blobs fetch nrf_wifi`.

Signed-off-by: Jordan Yates <jordan@embeint.com>

rruuaanng committed:
dts: bindings: dac: Change the property names in the overlay

Change the property names in the bindings and overlay
to use hyphens(-) for separation instead of underscores(_).

Signed-off-by: James Roy <rruuaanng@outlook.com>

jukkar committed:
net: pkt: Alloc headroom also for variable size data buffers

The headroom was not taken into account for variable size data
buffers when CONFIG_NET_L2_ETHERNET_RESERVE_HEADER was enabled.

Add a test case for it to make sure the reserve allocation works
properly.

Fixes #84053

Signed-off-by: Jukka Rissanen <jukka.rissanen@nordicsemi.no>

mrodgers-witekio committed:
net: http_server: fix URL matching with '?' character in resource

Fixes #84198.

If a '?' character is used as part of a wildcard resource, do not treat
this as the end of the string when comparing with a path from the HTTP
request. Only the path from the HTTP request may be terminated by '?'
(in the case of a request with query parameters).

Signed-off-by: Matt Rodgers <mrodgers@witekio.com>

ofirshe committed:
dts: nxp: rt1060: correct PTP clock reference in enet2

The enet2 node in nxp_rt1060.dtsi incorrectly references &enet_ptp_clock
for its nxp,ptp-clock property. This commit updates the reference to
&enet2_ptp_clock.

Signed-off-by: Ofir Shemesh <ofirshemesh777@gmail.com>

nordicjm committed:
cmake: mcuboot: Use zephyr-discovered imgtool

Uses imgtool which has already been located by zephyr

Signed-off-by: Jamie McCrae <jamie.mccrae@nordicsemi.no>

rlubos committed:
net: coap: Fix coap_packet_is_request() check for empty code

Empty code was incorrectly matched as a request, fix that.

Align coap_handle_request_len() function to behave as documented in the
API documentation - in case of invalid request code (which is also the
case for empty code) -ENOTSUP Should be returned.

Signed-off-by: Robert Lubos <robert.lubos@nordicsemi.no>

nordic-krch committed:
tests: drivers: uart: uart_mix_fifo_poll: Enable device PM

uart120 requires device runtime to be enable. Enable it for nrf54h20dk.

When device runtime PM is used for interrupt driven and polling API
then UART device is initially suspended. It means that RX is disabled.
In order to enable RX device must be explicitly resumed using PM API.

Test is enabling UART RX (uart_rx_enable) from counter callback
(interrupt handler context). For fast instance on nrf54h20dk (uart120)
it is not allowed because PM resume can only be called from the thread
context. Because of that, test is skipped for uart120 and asynchronous
API.

Signed-off-by: Krzysztof Chruściński <krzysztof.chruscinski@nordicsemi.no>

nordic-krch committed:
tests: drivers: uart: uart_mix_fifo_poll: Move counter handling

Move enabling of counter to the test instead of test setup. Test may be
skipped in some configurations and in that case counter shall not be
started so by moving setup to the test code allows skipping test before
counter is started.

Signed-off-by: Krzysztof Chruściński <krzysztof.chruscinski@nordicsemi.no>

nordic-krch committed:
tests: drivers: uart: uart_mix_fifo_poll: Clean up testcase.yaml

Clean up after automatic merge. Some test configurations got removed
and multiple lines enabling test for nrf54l09pdk/nrf54l09/cpuapp
target remained at the bottom of the file.

Signed-off-by: Krzysztof Chruściński <krzysztof.chruscinski@nordicsemi.no>

jukkar committed:
net: ipv4: Do not change the protocol type when sending

The ARP code has set the protocol type of the packet to
0x806, so do not change it when preparing to send to 0x800
which is the IP protocol type. Lets trust the previously
called functions to set the ptype correctly and do not set
it here.

Signed-off-by: Jukka Rissanen <jukka.rissanen@nordicsemi.no>

nrbrook committed:
cmake: Added "-" support to EXTRAVERSION

cmake: Added "-" support to EXTRAVERSION.
Added to the regex and update the documentation.
Also updated the examples in the documentation.

Signed-off-by: Nick Brook <nrbrook@gmail.com>

jukkar committed:
net: ipv6: Do not set ptype when preparing for sending

Trust that the protocol type is set correctly by functions
called before this one. We should not set the protocol type
blindly in this generic function.

Signed-off-by: Jukka Rissanen <jukka.rissanen@nordicsemi.no>

jacob-wienecke-nxp committed:
samples: net: zperf: Add correct .conf file name for mimxrt1060_evk

Change the name of the mimxrt1060_evk conf file (mimxrt1060_evk.conf) to
the correct name, "mimxrt1060_evk_mimxrt1062_qspi.conf". This change
causes net related code to be placed into ITCM on mimxrt1060_evk boards

Signed-off-by: Jacob Wienecke <jacob.wienecke@nxp.com>

Thalley committed:
tests: Bluetooth: Tester: Reorder btp_buf to fix variable array

The `btp_hdr` needs to be last in the struct, as it has
a variable length array (data[]).

Signed-off-by: Emil Gydesen <emil.gydesen@nordicsemi.no>

Commit:
doc: nxp: nw612: clarify documentation

NW612 is the firmware name, IW612 is the chipset name.

Signed-off-by: Derek Snell <derek.snell@nxp.com>

paulwedeck committed:
drivers: uart_wch: Fix compiler warning

This commit removes an unused variable.

Signed-off-by: Paul Wedeck <paulwedeck@gmail.com>

rruuaanng committed:
doc: Add a new migration guide entry to DAC

Rename the voltage_reference and power_down_mode
properties to voltage-reference and power-down-mode
in the devicetree and bindings of the DAC subsystem.

Signed-off-by: James Roy <rruuaanng@outlook.com>

TomasGalbickaNXP committed:
soc: Enable reset hook for mcxn236

This commit selects reset hook for mcxn236 in Kconfig.

This fixes the issue #84213.

Signed-off-by: Tomas Galbicka <tomas.galbicka@nxp.com>

jukkar committed:
net: Set the protocol type of fragmented packet

The fragmented packet should inherit the protocol type of the
original packet.

Signed-off-by: Jukka Rissanen <jukka.rissanen@nordicsemi.no>

jukkar committed:
net: Build assert issue with llvm

Remove the build assert from NET_L3_REGISTER() macro as that
is causing an issue with llvm. Add runtime check of the handler
pointer value.

subsys/net/l2/ethernet/arp.c:1044:1: error: static_assert expression
is not an integral constant expression

ETH_NET_L3_REGISTER(ARP, NET_ETH_PTYPE_ARP, arp_recv);
^~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
include/zephyr/net/ethernet.h:1272:2: note: expanded from
macro 'ETH_NET_L3_REGISTER'
        NET_L3_REGISTER(&NET_L2_GET_NAME(ETHERNET), name, ptype, handler)
        ^~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
include/zephyr/net/net_core.h:190:2: note: expanded from
macro 'NET_L3_REGISTER'
        BUILD_ASSERT((_handler) != NULL, "Handler is not defined")
        ^~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
include/zephyr/toolchain/gcc.h:87:51: note: expanded from
macro 'BUILD_ASSERT'
define BUILD_ASSERT(EXPR, MSG...) _Static_assert((EXPR), "" MSG)
                                                  ^~~~~~
subsys/net/l2/ethernet/arp.c:1044:1: note: cast from 'void *' is not
allowed in a constant expression
include/zephyr/net/ethernet.h:1272:2: note: expanded from
macro 'ETH_NET_L3_REGISTER'
        NET_L3_REGISTER(&NET_L2_GET_NAME(ETHERNET), name, ptype, handler)
        ^
include/zephyr/net/net_core.h:190:29: note: expanded from
macro 'NET_L3_REGISTER'
        BUILD_ASSERT((_handler) != NULL, "Handler is not defined")
                                   ^
/usr/lib/llvm-14/lib/clang/14.0.0/include/stddef.h:89:16: note: expanded
from macro 'NULL'
  define NULL ((void*)0)

Signed-off-by: Jukka Rissanen <jukka.rissanen@nordicsemi.no>

wwhheerree committed:
samples: subsys: shell: shell_module: Incorrect help message of sub_cmd1

Change sub_cmd1 help message from "help for cmd2" to "help for sub_cmd1".

Signed-off-by: Rick Cai <wwhheerree@hotmail.com>

jukkar committed:
net: ethernet: Properly handle VLAN tag 0

Packets are forwarded to the native interface or in other words,
the vlan header is simply stripped and ignored. This feature is called
'priority tagging'.

Signed-off-by: Cla Mattia Galliard <cla-mattia.galliard@zuehlke.com>
Signed-off-by: Jukka Rissanen <jukka.rissanen@nordicsemi.no>

robi251 committed:
drivers: dm8806: disable ethernet efficient mode

Energy efficient mode is the feature of the DM8806 described in the
EEE 802.3az Energy Efficient for reducing power consumption. For unknown
reason it seams to not working correctly with all endpoints and sometimes
the network randomly fails when this mode is on. Thus it is now possible to
turn off it in compile time by KConfig option:
PHY_DM8806_ENABLE_ENERGY_EFFICIENT_MODE if in case of network problems

Signed-off-by: Robert Slawinski <robert.slawinski1@gmail.com>

pdgendt commented on drivers/ethernet/phy/Kconfig.dm8806:
```diff
@@ -29,6 +29,14 @@ config PHY_DM8806_TRIGGER_GLOBAL_THREAD
 
 endchoice
 
+config PHY_DM8806_ENABLE_ENERGY_EFFICIENT_MODE
+	bool "802.3az Energy efficient mode"
+	default n
```
Omit `default n`, and I wouldn't put `ENABLE` in a boolean option.

```suggestion
config PHY_DM8806_ENERGY_EFFICIENT_MODE
	bool "802.3az Energy efficient mode"
```

robi251 replied:
I will take into account this remark. But because of conflicts with different fork (https://github.com/zephyrproject-rtos/zephyr/pull/84298), I would like to avoid a mess and I will remove this PR and create another one with your suggestion. Unless there is another way to resolve this conflict?

pdgendt replied:
Where is the conflict? Simply modify your branch of this PR locally and force push?

tmilkovic51 committed:
drivers: can: can_tcan4x5x: fix compiler build warning/error

Fix compiler warning when optional property reset-gpios
is not supplied in the ti,tcan4x5x-compatible device tree
node

Signed-off-by: Tomislav Milkovic <tomislav.milkovic95@gmail.com>

jukkar committed:
tests: net: vlan: Add VLAN tag 0 test

Make sure we are able to receive VLAN tag 0 packet, and
verify that reply packet is sent correctly.

Signed-off-by: Jukka Rissanen <jukka.rissanen@nordicsemi.no>

Commit:
arch: x86: add missing semicolon for z_x86_kpti_is_access_ok()

The function prototype / extern declaration for
z_x86_kpti_is_access_ok() in x86_mmu.h was missing a semicolon.

Add it to avoid being surprised by compile errors in certain
circumstances.

Signed-off-by: Chris Friedt <cfriedt@tenstorrent.com>

npitre committed:
kernel/pipe: squash compiler warning

The compiler complains that:

```
zephyr/kernel/include/kernel_internal.h:121:29:
error: 'reader' may be used uninitialized [-Werror=maybe-uninitialized]
  121 |         thread->swap_retval = value;
      |         ~~~~~~~~~~~~~~~~~~~~^~~~~~~
zephyr/kernel/pipe.c: In function 'copy_to_pending_readers':
zephyr/kernel/pipe.c:92:26: note: 'reader' was declared here
   92 |         struct k_thread *reader;
      |                          ^~~~~~
```

The static analyzer fails to see through the `LOCK_SCHED_SPINLOCK`
construct that the `reader` pointer is always initialized.

Signed-off-by: Nicolas Pitre <npitre@baylibre.com>

pdgendt commented:
Do not add merge commits, rebase your branch onto main and force push.

robi251 commented:
Pull request close due to few mistakes introduced during rebase.
